Kudu.Core/Jobs/JobsManagerBase.cs

```diff
@@ -495,6 +496,16 @@ private static string FindCommandToRun(FileInfoBase[] files, out IScriptHost scr
                         }
                     }
                 }
+
+                foreach (var supportedFileName in scriptHost.SupportedFileNames)
```

Comment:
Added a test case in bb94ee0c9d41eade914b747fa56aa73094c3a239. Also updated [`JobsScriptsShouldBeChosenByOrder()`](https://github.com/projectkudu/kudu/blob/bb94ee0c9d41eade914b747fa56aa73094c3a239/Kudu.FunctionalTests/Jobs/WebJobsTests.cs#L715) to look for `project.json`